src/Package/Impl/ProjectSystem/Commands/CopyItemPathCommand.cs

```diff
@@ -36,19 +37,20 @@ internal sealed class CopyItemPathCommand : IAsyncCommandGroupHandler {
         }
 
         public async Task<bool> TryHandleCommandAsync(IImmutableSet<IProjectTree> nodes, long commandId, bool focused, long commandExecuteOptions, IntPtr variantArgIn, IntPtr variantArgOut) {
-            if (commandId == RPackageCommandId.icmdCopyItemPath) {
-                var path = nodes.GetSingleNodePath();
-                var directory = await _interactiveWorkflowProvider.GetOrCreate().RSession.MakeRelativeToRUserDirectoryAsync(path);
-                if (!string.IsNullOrEmpty(directory)) {
-                    await VsAppShell.Current.DispatchOnMainThreadAsync(() => {
-                        try {
-                            Clipboard.SetData(DataFormats.UnicodeText, Invariant($"\"{directory}\""));
-                        } catch (ExternalException) { }
-                    });
-                }
-                return true;
+            VsAppShell.Current.AssertIsOnMainThread();
```

Comment:
Yeah, me too. But `InternalExecCommandAsync` which calls all the handlers is called from either `IOleCommandTarget.Exec` or `IVsUIHierarchy.ExecCommand`. AFAIK, both are called in from UI thread